src/cpu/aarch64/acl_gemm_convolution.cpp

```diff
@@ -14,30 +14,142 @@
+    using namespace data_type;
+    using smask_t = primitive_attr_t::skip_mask_t;
+
+    bool ok = is_fwd() && set_default_alg_kind(alg_kind::convolution_direct)
+            && expect_data_types(src_t, wei_t, bia_t, dst_t, undef)
+            && !has_zero_dim_memory()
+            && attr()->has_default_values(
+                    smask_t::post_ops | smask_t::fpmath_mode, dst_t)
+            && output_scales_mask_ok() && zero_points_ok();
+    if (!ok) return status::unimplemented;
+
+    CHECK(init_conf());
+
+    Op conv;
+    conv.configure(&acp_.src_tensor_info, &acp_.wei_tensor_info,
+            acp_.with_bias ? &acp_.bia_tensor_info : nullptr,
+            &acp_.dst_tensor_info, acp_.padstride_info, acp_.weights_info,
+            acp_.dilation_info, acp_.act_info, acp_.fast_math);
+
+    auto scratchpad = scratchpad_registry().registrar();
+    const auto mem_req = conv.workspace();
+    return init_scratchpad(conv, scratchpad, gemm_conv_keys, engine, post_ops,
+            attr_.post_ops_, acp_.act_info, acp_.use_dst_acc_for_sum, dst_md_);
+}
+
+template <data_type_t src_t, data_type_t wei_t, data_type_t dst_t,
+        data_type_t bia_t>
+status_t acl_gemm_convolution_fwd_t<src_t, wei_t, dst_t, bia_t>::init(
+        engine_t *engine) {
+    auto acp_ = pd()->acp_;
+    acl_obj_->conv.configure(&acp_.src_tensor_info, &acp_.wei_tensor_info,
+            acp_.with_bias ? &acp_.bia_tensor_info : nullptr,
+            &acp_.dst_tensor_info, acp_.padstride_info, acp_.weights_info,
+            acp_.dilation_info, acp_.act_info, acp_.fast_math);
+    acl_obj_->aux_mem_req = acl_obj_->conv.workspace();
+    return status::success;
+}
+
+template <data_type_t src_t, data_type_t wei_t, data_type_t dst_t,
+        data_type_t bia_t>
+status_t
+acl_gemm_convolution_fwd_t<src_t, wei_t, dst_t, bia_t>::execute_forward(
+        const exec_ctx_t &ctx) const {
+    return execute_forward_conv_acl<acl_obj_t<Op>, pd_t, src_data_t, wei_data_t,
+            dst_data_t, bia_data_t>(ctx, acl_obj_.get(), pd(), gemm_conv_keys);
+}
+
+template <data_type_t src_t, data_type_t wei_t, data_type_t dst_t,
+        data_type_t bia_t>
+status_t
+acl_gemm_convolution_fwd_t<src_t, wei_t, dst_t, bia_t>::pd_t::init_conf() {
+    if (weights_md_.ndims != 4) return status::unimplemented;
+
+    // General Compute Library checks, memory tags are also set there
+    CHECK(acl_convolution_utils::acl_init_conf(
+            acp_, src_md_, weights_md_, dst_md_, bias_md_, *desc(), *attr()));
+
+    // clang-format off
+    // Validate convolution manually to check for return status
+    ACL_CHECK_VALID(Op::validate(
+        &acp_.src_tensor_info,
+        &acp_.wei_tensor_info,
+        acp_.with_bias ? &acp_.bia_tensor_info : nullptr,
+        &acp_.dst_tensor_info,
+        acp_.padstride_info,
+        acp_.weights_info,
+        acp_.dilation_info,
+        acp_.act_info,
+        acp_.fast_math));
+    // clang-format on
+
+    return status::success;
```

Comment:
Optional: this would probably be clearer just sat straight in pd_t::init. It was originally separate because it was (somewhat) common and in conv utils.